docs/content/languages/java.md

```diff
@@ -0,0 +1,226 @@
+---
+title: "Using Java with Fission JVM environment"
+weight: 10
+---
+
+Fission supports functions written in Java and runs then on JVM. Current JVM environment is based on openjdk8 and uses Spring boot as framework.
+
+## Before you start
+
+We'll assume you have Fission and Kubernetes setup.  If not, head over
+to the [install guide](../installation/_index.en.md).  Verify your Fission setup with:
+
+```
+$ fission --version
+```
+
+## Add JVM environment to your cluster
+
+Fission language support is enabled by creating an _Environment_.  An environment is the language-specific part of Fission.  It has a container image in which your function will run.
+
+```
+$ fission environment create --name java --image fission/jvm-env --builder fission/jvm-builder
+```
+
+## Write a simple function in Java
+
+A function needs to implement the `io.fission.Function` class and override the `call` method. The call method receives the `RequestEntity` and `Context` as inputs and needs to return `ResponseEntity` object. Both `RequestEntity` and `ResponseEntity` are from `org.springframework.http` package and provide a fairly high level and rich API to interact with request and response objects. The function code responds with "Hello World" in response body looks as shown below:
+
+```java
+package io.fission;
+
+import org.springframework.http.RequestEntity;
+import org.springframework.http.ResponseEntity;
+
+import io.fission.Function;
+import io.fission.Context;
+
+public class HelloWorld implements Function {
+
+	@Override
+	public ResponseEntity<?> call(RequestEntity req, Context context) {
+		return ResponseEntity.ok("Hello World!");
+	}
+
+}
+```
+
+## HTTP requests and HTTP responses
+
+Java function provides easy access to the Request and Response using Spring framework's `RequestEntity` and `ResponseEntity` objects.
+
+### Accessing HTTP Requests
+
+#### Headers
+
+You can access headers object from the request object and then use various methods on header object to retrieve a specific header or get a collection of all headers.
+
+```java
+	HttpHeaders headers = req.getHeaders();
+	List<String> values = headers.get("keyname");
```

Comment:
Not sure whether nodejs handles canonical MIME format header key or not, but better to follow its format.
Details: https://github.com/fission/docs.fission.io/pull/78/files#diff-4901c14806cbf5768260a3cdeff0ab92R127

Reply:
I think you mean Java - I don't think this is supported in Java yet.

Reply:
I mean we should let users know the key will be converted to canonical MIME format header key.
For example, if a client passes "x-test-foo", then in function they need to use "X-Test-Foo" to get the value.